### PR TITLE
[feat] add ddpg, clean some code, modify algo_wrapper

### PR DESCRIPTION
**Pull request by zmsn-2077**
## Description

1. add off-policy ddpg and test it in SafetyPointGoal1
2. rename train_on_policy.py to train_policy.py, unify off-policy and on-policy api
3. del some needless code
4. on-policy code pass pylint
5. some spelling mistakes were corrected

image

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

**Review by hdadong — APPROVED** [review]

**Review by hdadong — APPROVED** [review]

**Review by Gaiejj — APPROVED** [review]

File: omnisafe/algos/configs/off-policy/DDPG.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+defaults:
+# Basic Configurations
+    epochs: 100
+    steps_per_epoch: 4000
```
Comment (XuehaiPan) [inline omnisafe/algos/configs/off-policy/DDPG.yaml:4]: This file is not pretty formatted.

File: omnisafe/algos/configs/on-policy/PPOLag.yaml [hunk 2]
```diff
@@ -14,58 +14,115 @@
...
-  max_ep_len: 1000
-  num_mini_batches: 16
-  pi_lr: 0.0003
-  vf_lr: 0.001
-  target_kl: 0.01
-  data_dir: "./runs"
-  seed: 0
+# Basic Configurations
+    epochs: 500
+    steps_per_epoch: 30000
```
Comment (XuehaiPan) [inline omnisafe/algos/configs/on-policy/PPOLag.yaml:19]: Please follow the `.editorconfig`: ident with 2 spaces.

File: omnisafe/algos/on_policy/cppo_pid.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-
+"""Implementation of the CPPOPid algorithm."""
```
Comment (XuehaiPan) [inline omnisafe/algos/on_policy/cppo_pid.py:15]: Please add an extra linebreak between the module docstring and imports. Same changes for other source files.

File: .gitignore [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ##### Project Specification #####
 **/runs
-
+**/scripts
```
Comment (XuehaiPan) [inline .gitignore:3]: Add an extra linebreak here.

File: omnisafe/algos/__init__.py [hunk 5]
```diff
@@ -12,9 +12,36 @@
...
+        'PCPO',
+        'PDO',
+        'PolicyGradient',
+        'PPO',
+        'PPOLag',
+        'TRPO',
+        'TRPOLag',
+    ],
+    'model-based': ['MBPPOLag', 'SafeLoop'],
+}
```
Comment (XuehaiPan) [inline omnisafe/algos/__init__.py:35]: Constants should use `ALL_CAPS` names. Also, this may place after the import statements.

**Review by XuehaiPan — COMMENTED** [review]
The commit message is not in semantic commit format.

**Pull request merged**